### PR TITLE
Fe feat/connect query detailspage

### PR DESCRIPTION
**Pull request by Elise-Jones**
# Description
- Connect data to the details page
- Use useLocation to connect individual mushrooms. 
Closes# (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] Testing
- [ ] Refactoring

# How Has This Been Tested?

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Components/DetailsPage/DetailsPage.js [hunk 1]
```diff
@@ -1,10 +1,16 @@
 import React from "react";
 import "./DetailsPage.css";
+import { useLocation} from "react-router-dom";
 
-const DetailsPage = ({ latinName, commonName, edibility, taxonomy, photo, characteristic, description, lookalike, psychoactive }) => {
```
Comment (jmartin777) [inline src/Components/DetailsPage/DetailsPage.js:4]: Awesome change here with he logic of the Details page!

File: src/Components/DetailsPage/DetailsPage.js [hunk 2]
```diff
@@ -30,24 +34,24 @@ const DetailsPage = ( ) => {
             </div>
             <div className="taxonomy">
               <h3>Characterisics:</h3>
-              <p> Hymenium Type: {characteristic.hymeniumType},</p>
-              <p> Stipe Character: {characteristic.stipeCharacter},</p>
-              <p> Spore Print Color {characteristic.sporePrintColor},</p>
-              <p> Mushroom Cap Shape: {characteristic.mushroomCapShape},</p>
-              <p> Hymenium Attachment: {characteristic.hymeniumAttachment}</p>
-              <p> Mushroom Ecological Type: {characteristic.mushroomEcologicalType}</p>
+              <p> Hymenium Type: {characteristics.hymeniumType},</p>
```
Comment (jmartin777) [inline src/Components/DetailsPage/DetailsPage.js]: reflects the query database now! awesome

**Review by jmartin777 — APPROVED** [review]
Great Job Elise on crushing this out! Was really awesome to see the details page in the meeting with Cass today!

File: src/Components/DetailsPage/DetailsPage.js (hunk 2 above)
Comment (andreasorensen) [reply src/Components/DetailsPage/DetailsPage.js]: So cool!!!

File: src/Components/App/App.js [hunk 3]
```diff
@@ -43,9 +43,10 @@ function App() {
 
     return data.mushrooms.map((mushroom) => (
       <div className="mushroom-card-wrapper">
-        <NavLink to={`details/${mushroom.id}`} className="custom-nav-link">
+        <NavLink to={`details/${mushroom.id}`} className="custom-nav-link" state={mushroom}>
```
Comment (andreasorensen) [inline src/Components/App/App.js:46]: Good job adding this mushroom state!

File: src/Components/App/App.js [hunk 4]
```diff
@@ -43,9 +43,10 @@ function App() {
...
           <MushroomCard
             key={mushroom.apiId}
+            id={mushroom.apiId}
```
Comment (andreasorensen) [inline src/Components/App/App.js:49]: I think this needs to be mushroom.id once we render for the field guide

File: src/Components/DetailsPage/DetailsPage.js [hunk 5]
```diff
@@ -1,46 +1,70 @@
 import React from "react";
 import "./DetailsPage.css";
+import { useLocation } from "react-router-dom";
```
Comment (andreasorensen) [inline src/Components/DetailsPage/DetailsPage.js:3]: YES!

**Review by andreasorensen — APPROVED** [review]
Great job getting this all connected!!!

**Pull request merged**